HelpSource/Guides/Tour-of-Special-Functions.schelp

```diff
@@ -0,0 +1,1073 @@
+TITLE:: Tour of Special Functions
+SUMMARY:: Examples and plots for Boost Special Functions
+CATEGORIES:: Guides
+related:: Classes/Number
+KEYWORD:: special functions, boost
+
+
+DESCRIPTION:: Boost library Special Functions: usage, bounds and plots
+
+An array of Special Functions supplied by the Boost library. The library's
+LINK::http://www.boost.org/doc/libs/1_66_0/libs/math/doc/html/special.html##online documentation::
+ serves as the primary reference for the following functions.
+
+Many of the functions are only valid in certain numerical ranges. The following tour will
+show you plots that illustrate the behavior of the functions, their ranges and asymptotes.
+
+Define the plotting functions in the next section, then you can use the Table of Contents
+menu to jump between function families.
+
+SECTION:: Before you get started...
+
+Define the following functions that will lay out the plots for you:
+
+CODE::
+(
+var genCol, getPlotLay, multiPlot, layOutPlots, plotSpecs;
+
+// generate colors for each plot
+genCol = { |num, offset=(rrand(0,1.0)), range=(rrand(0.15,1.3))|
+	num.collect{|i| Color.hsv(((i/(max(num,1.0001)-1)) * range + offset).wrap(0,1), 0.6, 1, 1)}
+};
+// generate a vertical layout for the plot and its text
+getPlotLay = { |plot, text|
+	VLayout(plot, StaticText().string_(text).maxHeight_(50).align_(\center), 10);
+};
+
+~multiPlot = { |func, data, txt, min, max|
+	var res, params, samps, pltr;
+	var gridcol = Color.gray.alpha_(0.3);
+
+	res = if (data.shape.size == 1) { // 1D data
+		params = [0];
+		samps = data;
+		samps.collect({ |smp|
+			if (func.isKindOf(Symbol)) {
+				smp.perform(func)
+			}{
+				func.(smp)
+			}
+		}).select({|me| me.notNil})
+	}
+	{ // 2D data
+		#params, samps = data;
+		txt = txt ++ params.asString;
+		params.collect({ |param, i|
+			samps.collect({ |smp|
+				if (func.isKindOf(Symbol)) {
```

Comment:
you can replace this code by: `func.applyTo(smp, i)`

Reply:
Useful! Done.